### ISSUE TITLE
Awaitable vs. Generator[Any, None, ResultType] in asyncio API's [bad-return-type]

### ISSUE THREAD

**Issue by byllyfish** [google/pytype#244]
Using pytype 2019.02.13 with python 3.7.1 and 3.6.5, when I use an asyncio API, I receive a bad-return-type warning:

Here's an example using the first few lines from the sample code (https://docs.python.org/3/library/asyncio-stream.html):

```python
import asyncio

async def tcp_echo_client(message):
    reader, writer = await asyncio.open_connection( '127.0.0.1', 8888)
```

Here are is the warning:

```
File "/test_awaitable1.py", line 4, in tcp_echo_client: bad option in return type [bad-return-type]
  Expected: Awaitable
  Actually returned: Generator[Any, None, Tuple[asyncio.streams.StreamReader, asyncio.streams.StreamWriter]]
```

The warnings seems to happen for all free async functions in asyncio. In typeshed, they are annotated to return the Generator type, but it's not coercing into an Awaitable.

Thanks!

**Comment by rchen152** [comment]
Hmm,  yeah, we definitely shouldn't be erroring on sample code =P Thanks for the report!

**Comment by ozankabak** [comment]
FYI, I just ran into this issue as well, but while using an `asyncio.Queue`, not a free-standing function.

**Comment by komuw** [comment]
my code is asyncio heavy and I just came across this

```python
async def my_number() -> int:
    return 56


async def ala() -> None:
    z = await my_number()
    print(z)
```

```sh
bad option in return type [bad-return-type]
  Expected: Awaitable
  Actually returned: Generator[Any, None, nothing]
```

**Comment by rchen152** [comment]
I'll take a look!

**Comment by rchen152** [comment]
I have a fix for this issue out for review now. Our release pipeline is running a little behind schedule due to a large upcoming change (attrs support!), so this fix probably won't land for another week or two. I'll leave this issue open until there's a PyPI release containing the fix.

**Comment by rchen152** [comment]
The fix was released in pytype 2019.06.21. Let me know if this problem still persists.
